pkg/interceptors/cel/cel.go

```diff
@@ -61,7 +61,7 @@ var (
 )
 
 // NewInterceptor creates a prepopulated Interceptor.
-func NewInterceptor(k kubernetes.Interface, l *zap.SugaredLogger) *Interceptor {
+func NewInterceptor(k kubernetes.Interface, l *zap.SugaredLogger) interceptors.Interceptor {
```

Comment:
I think we want to still return *Interceptor here - by doing this you're masking the `triggersv1.InterceptorInterface` and other methods which limits how this can be used by callers.

If you want static type checking use a similar trick as the `triggersv1.InterceptorInterface` var above.

Reply:
If you do this I think you can also get rid of the type assertion in server.go below.

Reply:
So, I added this for this to be consistent with the other interceptors which all return interceptors.Interceptor (i.e. the old interface). I could flip it to return the new interface but that will be a slightly larger change -- we'll have to change some type assertions in sink.go since the webhook interceptor will still be using the old interface at the moment.

Reply:
ok, switched all of the interceptors to the new interface.